numpy/core/tests/test_multiarray.py

```diff
@@ -5006,7 +5007,19 @@ def test_basic(self):
             self._clip_type(
                 'uint', 1024, 0, 0, inplace=inplace)
             self._clip_type(
-                'uint', 1024, -120, 100, inplace=inplace, expected_min=0)
+                'uint', 1024, 10, 100, inplace=inplace)
+
+    @pytest.mark.parametrize("inplace", [False, True])
+    def test_int_range_error(self, inplace):
+        # E.g. clipping uint with negative integers fails to promote
+        # (changed with NEP 50 and may be adaptable)
```

Comment:
There are possible solution, but right now they need a fair bit of complexity (maybe some adaptation too, but not much).  I think we need a ufunc implementation for `(any_integer, pyinteger)` and `(pyinteger, anyinteger)`.
And either we allow access to the scalar values (if an abstract scalar) in one of the steps (promotion or loop selection), or (which is possible now) we put it into the inner-loop itself.

I.e. you can effectively make it a loop that uses the Python object for the scalar (and then deal with the rest in the loop itself).

That ain't pretty, although fortunately it would only be necessary for comparisons.

(maybe there are other ways to tweak it to be a bit more reasonable, not sure)